File: web-client/www/src/js/representation/invitation.representation.js
/**
 * @typedef {"CREATED"|"USED"|"REVOKED"} InvitationLinkStatus
 * @typedef {"APPLICATION_JOIN"|"SERVER_JOIN"} InvitationType
 */

class InvitationRepresentation {
  /** @type {string} */
  id
  /** @type {InvitationLinkStatus} */
  status
  /** @type {InvitationType} */
  type
  /** @type {string} */
  targetedServer
}

File: core-server/app/src/main/java/fr/revoicechat/core/representation/InvitationRepresentation.java
package fr.revoicechat.core.representation;

import java.util.UUID;

import org.eclipse.microprofile.openapi.annotations.media.Schema;

import fr.revoicechat.core.model.InvitationLinkStatus;
import fr.revoicechat.core.model.InvitationType;

@Schema(name = "InvitationRepresentation",
    description = "Data returned when an invitation is created")
public record InvitationRepresentation(UUID id,
                                       InvitationLinkStatus status,
                                       InvitationType type,
                                       UUID targetedServer) {}
